test/endpoint.jl

```diff
@@ -249,6 +249,9 @@ using Intervals: LeftEndpoint, RightEndpoint
 
         @test 1.0 != Endpoint(2, false)
         @test 1.0 != Endpoint(2, true)
+
+        result = 1.0 .== [Endpoint(1, false), Endpoint(1, true)]
+        @test result == [false, true]
```

Comment:
Should have a comment:

> Verify that `Endpoint` is treated as a scalar during broadcast

Reply:
Without the comment I think I feel like I would refactor this code and possibly not test what was intended